src/prefs/DirectoriesPrefs.cpp

```diff
@@ -447,18 +467,40 @@ bool DirectoriesPrefs::Validate()
          wxOK | wxCENTRE | wxICON_INFORMATION);
    }
 
+   const wxString openPathString = mOpenText->GetValue();
+   const wxString savePathString = mSaveText->GetValue();
+   const wxString importPathString = mImportText->GetValue();
+   const wxString exportPathString = mExportText->GetValue();
    const wxString macroPathString = mMacrosText->GetValue();
+   const std::initializer_list<TranslatableString> messagesPreference{
+      XO("'Open' directory cannot be set.") ,
+      XO("'Save' directory cannot be set.") ,
+      XO("'Import' directory cannot be set.") ,
+      XO("'Export' directory cannot be set.") ,
+      XO("'Macro Output' directory cannot be set.") ,
+   };
 
-   if (!macroPathString.empty())
-   {
-      const wxFileName macroPath { macroPathString };
-
-      if (macroPath.DirExists())
-      {
-         if (!FileNames::WritableLocationCheck(macroPathString))
-            return false;
+   //flag for checking if at least one directory write protected 
+   //will not be 0 if any of the paths are not writable
+   int flag = 0;
+   //id for indexing error messages to the initializer_list.
+   int id = 0;
+   //Checks if the location for Open,Save.Import,Export and Macro Output has write permissions(When OK is clicked)
+   for (auto &string : { openPathString, savePathString, importPathString, exportPathString, macroPathString} ) {
+      const wxFileName currentPath { string };
+      if(!string.empty()){
+         if (currentPath.DirExists()){
+            if(!FileNames::WritableLocationCheck(string, *(messagesPreference.begin() + id)))
+               flag++;
+         }
+         else {
+            return CreateDirectory(string, *(messagesPreference.begin() + id));
```

Comment:
`const auto & message = *(messagesPreference.begin() + id);` might come before the if-else so you can eliminate a repeated exppression.